nextflow.config

```diff
@@ -0,0 +1,595 @@
+trace {
+    enabled            = true
+    file               = "${params.outdir}/pipeline_info/pipeline_trace_${trace_timestamp}.txt"
+    fields             = 'task_id,name,process,tag,status,exit,module,time,disk,memory,attempt,start,complete,duration,realtime,queue,%cpu,%mem,rss,workdir,error_action'
+}
+timeline {
+    enabled            = true
+    file               = "${params.outdir}/pipeline_info/pipeline_timeline_${trace_timestamp}.txt"
+}
+report {
+    enabled            = true
+    file               = "${params.outdir}/pipeline_info/pipeline_report_${trace_timestamp}.html"
+}
+
+
+// process specific configs
+process {
+
+    withName: 'GUNZIP_*' {
+        publishDir = [
+            path: { "${params.outdir}/04_star/genome" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+            enabled: params.save_reference
+        ]
+    }
+
+    withName: 'SAMPLESHEET_CHECK' {
+        publishDir = [
+            path: { "${params.outdir}/pipeline_info" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
+    withName: 'DECOMP' {
+        publishDir = [
+            path: { "${params.outdir}/00_reads/unzipped" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+            pattern: '*.fastq'
+        ]
+    }
+
+    withName: 'COMP' {
+        publishDir = [
+            path: { "${params.outdir}/00_reads/zipped" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+            pattern: '*.fastq.gz'
+        ]
+    }
+
+    withName: 'REMOVE_CHAR' {
+        ext.cmd1 = "sed 's/ /#/g; s/\\//#/g' "
+        ext.suffix = '.edited'
+        ext.ext = 'fastq'
+        publishDir = [
+            path: { "${params.outdir}/00_reads/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+            mode: params.publish_dir_mode,
+            enabled: params.save_edited_reads
+        ]
+    }
+
+    withName: 'FASTQ_FASTQC_UMITOOLS_TRIMGALORE:FASTQC' {
+        ext.args   = '--quiet'
+        publishDir = [
+            path: { "${params.outdir}/01_${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+            mode: params.publish_dir_mode
+        ]
+    }
+
+// some adapters from illumina: https://support.illumina.com/content/dam/illumina-support/help/Illumina_DRAGEN_Bio_IT_Platform_v3_7_1000000141465/Content/SW/Informatics/Dragen/FastQC_Adapter_Kmer_files_fDG.htm
+    withName: 'FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE' {
+        ext.args  = {
+            [
+                "--fastqc_args '-t ${task.cpus}'",
+                "-a 'AGATCGGAAGAGCGGTTCAG -a AGATCGGAAGAG -a TGGAATTCTCGG -a AAAAAAAAAAAA -a GGGGGGGGGGGG' ",
```

Comment:
Are we sure these are the adapters that will be used? Shouldn't this be a user input?